### PR TITLE
Enhancements for calculating water depth processing algorithm.

### PR DESCRIPTION
**Pull request by ldebek**
Modifications in processing tool for calculating water depth/level.
image

### CODE REVIEW

File: CHANGES.rst [hunk 1]
```diff
@@ -23,6 +23,7 @@ ThreeDiToolBox changelog
   installation) are made of all pull requests and of master. See
   https://docs.3di.live/threeditoolbox-dev/ .
 
+- Enhancements for water depth/level calculation processing tool
```
Comment (reinout) [inline CHANGES.rst]:
```suggestion
- Enhancements for the water depth/level calculation processing tool.
```

File: processing/threedidepth_algorithm.py [hunk 2]
```diff
@@ -14,14 +14,20 @@
...
+    QgsProcessingException,
+    QgsProcessingParameterBoolean,
+    QgsProcessingParameterEnum,
+    QgsProcessingParameterFile,
+    QgsProcessingParameterNumber,
+    QgsProcessingParameterRasterLayer,
+    QgsProcessingParameterString,
+    QgsRasterLayer,
+    QgsMeshLayer,
+)
```
Comment (reinout) [inline processing/threedidepth_algorithm.py]: This is not the import style that we want.

- We want to be able to use grep, single imports on a single line help with that.
- "make beautiful" also runs "isort", which will change your style back :-)

File: processing/threedidepth_algorithm.py [hunk 3]
```diff
@@ -321,6 +330,21 @@ def processAlgorithm(self, parameters, context, feedback):
...
+        raster_filename = parameters[self.WATER_DEPTH_LEVEL_NAME]
+        if not raster_filename:
+            raise QgsProcessingException(self.invalidSourceError(parameters, self.WATER_DEPTH_LEVEL_NAME))
+
+        output_location = parameters[self.OUTPUT_DIRECTORY]
+        if not output_location:
+            raise QgsProcessingException(self.invalidSourceError(parameters, self.OUTPUT_DIRECTORY))
+
+        as_netcdf = parameters[self.AS_NETCDF_INPUT]
+        raster_extension = "nc" if as_netcdf else "tif"
```
Comment (reinout) [inline processing/threedidepth_algorithm.py:340]: `tif` or `tiff`? I don't know what's used most often?

**Review by reinout — APPROVED** [review]
Looks fine, apart from the import style :-)

File: processing/threedidepth_algorithm.py (hunk 3 above)
Comment (ldebek) [reply processing/threedidepth_algorithm.py:340]: Raster format is `TIFF`, extension is `.tif`. :)

File: CHANGES.rst (hunk 1 above)
Comment (ldebek) [reply CHANGES.rst]: Done.

File: processing/threedidepth_algorithm.py (hunk 2 above)
Comment (ldebek) [reply processing/threedidepth_algorithm.py]: Done.

**Pull request merged**